axelor-stock/src/main/java/com/axelor/apps/stock/service/MassStockMoveServiceImpl.java

```diff
@@ -0,0 +1,130 @@
+import com.axelor.i18n.I18n;
+import com.axelor.inject.Beans;
+import com.google.inject.Inject;
+import com.google.inject.persist.Transactional;
+import java.math.BigDecimal;
+import java.util.List;
+import javax.swing.JOptionPane;
+
+public class MassStockMoveServiceImpl implements MassStockMoveService {
+
+  protected MassStockMoveRepository massStockMoveRepository;
+  protected SequenceService sequenceService;
+  protected PickedProductsRepository pickedProductRepo;
+
+  @Inject
+  public MassStockMoveServiceImpl(
+      MassStockMoveRepository massStockMoveRepository,
+      SequenceService sequenceService,
+      PickedProductsRepository pickedProductRepo) {
+    this.massStockMoveRepository = massStockMoveRepository;
+    this.sequenceService = sequenceService;
+    this.pickedProductRepo = pickedProductRepo;
+  }
+
+  @Override
+  @Transactional(rollbackOn = Exception.class)
+  public void importProductFromStockLocation(MassStockMove massStockMove) throws AxelorException {
+
+    try {
+
+      List<StockLocationLine> stockLocationLineList =
+          Beans.get(StockLocationLineRepository.class)
+              .all()
+              .filter("self.stockLocation = ?1", massStockMove.getCommonFromStockLocation())
+              .fetch();
+      List<StockLocationLine> detailsStockLocationLineList =
+          Beans.get(StockLocationLineRepository.class)
+              .all()
+              .filter("self.detailsStockLocation =?1", massStockMove.getCommonFromStockLocation())
+              .fetch();
+      if (stockLocationLineList.isEmpty() && detailsStockLocationLineList.isEmpty()) {
+        JOptionPane.showMessageDialog(
+            null,
+            "No stock location lines have been found in the common from stock location.",
+            "Warning",
+            JOptionPane.WARNING_MESSAGE);
+        return;
+      }
+      for (StockLocationLine line : stockLocationLineList) {
+        if (line.getProduct().getTrackingNumberConfiguration() == null
+            && line.getCurrentQty().compareTo(BigDecimal.ZERO) == 1) {
+          PickedProducts pickedProduct = new PickedProducts();
+          pickedProduct.setPickedProduct(line.getProduct());
+          pickedProduct.setTrackingNumber(null);
+          pickedProduct.setFromStockLocation(line.getStockLocation());
+          pickedProduct.setCurrentQty(line.getCurrentQty());
+          pickedProduct.setUnit(line.getProduct().getUnit());
+          pickedProduct.setMassStockMove(massStockMove);
+          massStockMove.getPickedProductsList().add(pickedProduct);
+          Beans.get(PickedProductsRepository.class).save(pickedProduct);
+        }
+      }
+      for (StockLocationLine line : detailsStockLocationLineList) {
+        if (line.getCurrentQty().compareTo(BigDecimal.ZERO) == 1
+            && line.getProduct() != null
+            && line.getProduct().getTrackingNumberConfiguration() != null) {
+          PickedProducts pickedProduct = new PickedProducts();
```

Comment:
Same as above, create/use new creation method for pickedProduct and use massStockMove.addPickedProductsListItem